### PR TITLE
calendar: move to "today" daily at midnight

### LINKED ISSUE #14235: Calendar does not update the day in dayGridDay
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.

### Describe the issue you are experiencing

With this card:
```
          - type: calendar
            entities:
              - calendar.family
              - calendar.her
              - calendar.him
              - calendar.holidays_in_united_states
            initial_view: dayGridDay
```
each day you have to hit the "today" button to see that day's events.

### Describe the behavior you expected

I expected that if the calendar is already displaying "today" that when the day changes, the card would be refreshed to the new day.

### Steps to reproduce the issue

1. Add the calendar card (see description)
2. Wait for tomorrow
3. See that the card's still showing yesterday
...

### What version of Home Assistant Core has the issue?

core-2023.2.0

### What was the last working version of Home Assistant Core?

None

### In which browser are you experiencing the issue with?

Android App

### Which operating system are you using to run this browser?

Android

### State of relevant entities

_No response_

### Problem-relevant frontend configuration

_No response_

### Javascript errors shown in your browser console/inspector

_No response_

### Additional information

The checkboxes:

- I have cleared the cache of my browser. *
- I have tried a different browser to see if it is related to my browser. * 

I don't believe these are applicable because I'm not using a browser. I only checked them so I could submit the issue. If I'm wrong, please provide guidance.

### PR DESCRIPTION
**Pull request by joejulian**
## Proposed change

Each day the calendar is showing the previous day and user action is required to change to today. This change is to allow the calendar to take that action itself at midnight.

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

fixes #14235

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Comment by karwosts** [comment]
Just wanted to check, did you want this to be reviewed, or is it still in development? You would need to take it out of draft mode before it will be considered.

**Comment by joejulian** [comment]
@karwosts I'm a novice when it comes to ts (or Javascript at all for that matter) so I at least wanted to find my way to a green CI before I marked it ready for review.

Is there any sort of tests that I should be providing with this? Does my approach seem sane?

**Comment by karwosts** [comment]
Unfortunately frontend doesn't have much test infrastructure, so I don't think there's anything much you could provide for testing. I'm still a bit of a novice myself so I can't say for certain if this is a good or bad approach. It seems like a sane implementation to me at first glance. I will let other members comment if this is something wanted for the frontend or not. 

I just wanted to confirm that you hadn't walked away forgetting about the draft setting.

**Comment by joejulian** [comment]
@bramkragten @ludeeus @piitaya Am I missing anything that I need to get a review? I looked at the contributing guidelines and I _think_ I should be good, but if there's something more that needs done, please let me know.

File: src/panels/calendar/ha-full-calendar.ts [hunk 1]
```diff
@@ -278,6 +278,17 @@ export class HAFullCalendar extends LitElement {
       this.shadowRoot!.getElementById("calendar")!,
       config
     );
+
+    // set the calendar to today every day at midnight
+    (function daily(calendar) {
+      const m = new Date().setHours(24, 0, 0, 0);
```
Comment (bramkragten) [inline src/panels/calendar/ha-full-calendar.ts:284]: You should take timezones into consideration, the timezone used in Home Assistant might be different than the timezone of the browser.

In general, set your timezone to something different than your browser is using:
CleanShot 2023-08-18 at 10 21 24

In profile, select use server timezone:
CleanShot 2023-08-18 at 10 22 42

File: src/panels/calendar/ha-full-calendar.ts [hunk 2]
```diff
@@ -278,6 +278,17 @@ export class HAFullCalendar extends LitElement {
       this.shadowRoot!.getElementById("calendar")!,
       config
     );
+
+    // set the calendar to today every day at midnight
+    (function daily(calendar) {
```
Comment (bramkragten) [inline src/panels/calendar/ha-full-calendar.ts:283]: Change this to:
```ts
const daily = () => {
	// and use `this.calendar`
}
```

File: src/panels/calendar/ha-full-calendar.ts (hunk 1 above)
Comment (joejulian) [reply src/panels/calendar/ha-full-calendar.ts:284]: Isn't the browser's "today" the one that matters though?

Comment (joejulian) [reply src/panels/calendar/ha-full-calendar.ts:284]: Where is `today()` defined? I can't find it.

```
  private _handleToday(): void {
    this.calendar!.today();
    this._fireViewChanged();
  }
```

Comment (bramkragten) [reply src/panels/calendar/ha-full-calendar.ts:284]: It is a function of Full Calendar

**Comment by frenck** [comment]
Because there hasn't been any activity on this PR for quite some time now, I've decided to close it for being stale.

Feel free to re-open this PR when you are ready to pick up work on it again 👍

**Comment by callumgare** [comment]
This is an incredibly ugly hack but for anyone looking for a workaround here's what I've done. I have [browser_mod](https://github.com/thomasloven/hass-browser_mod) installed and I use an automation to run the service browser_mod.javascript every night which executes some js to find and click the "Today" button on the calendar. As I say, very ugly! And I hope the this PR gets fixed and merged eventually but but until then this it works for me.

YAML for the automation:
```yaml
alias: Dashboard - update calendar to today view
description: ""
trigger:
  - platform: time
    at: "00:01:00"
condition: []
action:
  - service: browser_mod.javascript
    data:
      browser_id: a1b2c3
      code: |
        function clickTodayButton(node, shadowRoots = []) {
            if (!node) return;
        
          const todayButton = node?.shadowRoot?.querySelector("mwc-button.today")
        
          if (todayButton) {
            return todayButton.click()
          }
        
            if (node.shadowRoot) {
                shadowRoots.push(node.shadowRoot);
            clickTodayButton(node.shadowRoot, shadowRoots)
            }
        
            node.childNodes.forEach(childNode => {
                if (childNode.nodeType === Node.ELEMENT_NODE) {
                    clickTodayButton(childNode, shadowRoots);
                }
            });
        }
        
        clickTodayButton(document);
    target:
      entity_id: entity.not_needed

mode: single
```

**Pull request closed without merging**